### PR TITLE
fix a few small bugs

### PR DESCRIPTION
**Pull request by mcalhoun**
## what
* Update incorrect provider config
* Remove unused variable
* Add `.terraform.lock.hcl` to the `.gitignore`

## why

* The main driver for this PR was that we previously had:

```hcl
required_providers {
    local = {
      source  = "hashicorp/aws"
      version = ">= 3.0"
    }
  }
}
```

This lead to a misconfiguration which caused the AWS provider not to function properly and lead to several hours of debugging.

### CODE REVIEW

File: examples/complete/variables.tf [hunk 1]
```diff
@@ -1,8 +1,3 @@
-variable "region" {
-  type        = string
-  description = "AWS Region"
-}
```
Comment (nitrocode) [inline examples/complete/variables.tf:4]: How come this is removed ?

**Review by nitrocode — APPROVED** [review]

File: examples/complete/variables.tf (hunk 1 above)
Comment (nitrocode) [reply examples/complete/variables.tf:4]: Oh I see, it's unused.

Comment (mcalhoun) [reply examples/complete/variables.tf:4]: It's not used anywhere (we don't use the AWS provider)

**Pull request merged**